packages/components/src/components/hds/stepper/nav/step.hbs

```diff
@@ -0,0 +1,80 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+{{! template-lint-disable require-context-role no-invalid-role }}
+{{! template-lint-disable require-presentational-children }}
+<li class={{this.classNames}} ...attributes role={{if this.isNavInteractive "presentation"}}>
+  {{#if this.isNavInteractive}}
+    <button
+      class="hds-stepper-nav__step-content hds-stepper-nav__step-btn"
+      id={{this._stepId}}
+      tabindex={{unless (eq this.status "active") "-1"}}
+      type="button"
+      role="tab"
+      aria-controls={{this.coupledPanelId}}
+      aria-selected={{if (eq this.status "active") "true" "false"}}
+      aria-disabled={{if (eq this.status "incomplete") "true" "false"}}
+      {{on "click" this.onStepChange}}
+      {{on "keyup" this.onKeyUp}}
+      {{this._setUpStep this.didInsertNode this.willDestroyNode}}
+    >
+      <div class="hds-stepper-nav__step-progress">
+        <Hds::Stepper::Step::Indicator
+          @text="{{this.stepNumber}}"
+          @status={{this.stepIndicatorStatus}}
+          @isInteractive={{true}}
+          class="hds-stepper-nav__step-indicator"
+        />
+      </div>
+      <div class="hds-stepper-nav__step-text">
+        <Hds::Text::Body class="hds-stepper-nav__step-title" @tag={{this.titleTag}} @size="200" @weight="semibold">
+          <span class="sr-only">{{this.statusSrOnlyText}}</span>
+          {{yield to="title"}}
+        </Hds::Text::Body>
+        <Hds::Text::Body
+          class="hds-stepper-nav__step-description"
+          @tag="div"
+          @size="100"
+          @weight="regular"
+          @color="regular"
+        >
+          {{yield to="description"}}
+        </Hds::Text::Body>
+      </div>
+    </button>
+  {{else}}
+    <div
+      class="hds-stepper-nav__step-content"
+      id={{this._stepId}}
+      aria-current={{if (eq this.status "active") "true" "false"}}
+      {{this._setUpStep this.didInsertNode this.willDestroyNode}}
+    >
+      <div class="hds-stepper-nav__step-progress">
+        <Hds::Stepper::Step::Indicator
```

Comment:
I think the step indicator should be `aria-hidden` when it is not interactive.. reasoning being- the ordered list handles announcing which item in the list you are on so the step number being available to screen readers is redundant information.

Reply:
Hmm I don't think we need to hide it for both cases. Reason being, when its not interactive, it is an ordered list and the index of each list item is announced and its understood that the order is meaningful.

For interactive ones, it is no longer a list but a set of tabs so the step number information would be useful.